File: java-server/src/main/java/de/davelee/personalman/server/model/UserHistoryEntry.java
package de.davelee.personalman.server.model;

import lombok.*;
import org.bson.types.ObjectId;

import java.time.LocalDate;

/**
 * Class to represent a particular entry in the history of a particular user in PersonalMan.
 * @author Dave Lee
 */
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
public class UserHistoryEntry {

    /**
     * A unique id for this history entry.
     */
    private ObjectId id;

    /**
     * The date that this history entry took place.
     */
    private LocalDate date;

    /**
     * The reason for this history entry.
     */
    private UserHistoryReason userHistoryReason;

    /**
     * A comment about this history - this could be the reason it was given.
     */
    private String comment;

}

File: server/src/users/models/userhistory.entry.ts
export class UserHistoryEntry {

    /**
     * The date that this history entry took place.
     */
    date: Date;

    /**
     * The reason for this history entry.
     */
    reason: string;

    /**
     * A comment about this history - this could be the reason it was given.
     */
    comment: string;

    constructor(date: Date, reason: string, comment: string) {
        this.date = date;
        this.reason = reason;
        this.comment = comment;
    }

    getUserHistoryReason(): string {
        return this.reason;
    }

    getComment(): string {
        return this.comment;
    }

    getDate(): Date {
        return this.date;
    }

}